test/functional/mempool_compatibility.py

```diff
@@ -23,6 +23,7 @@ class MempoolCompatibilityTest(BitcoinTestFramework):
     def set_test_params(self):
         self.num_nodes = 2
         self.wallet_names = [None]
+        self.requires_wallet = True
```

Comment:
I can't understand why do we need "the old versions have their wallets' setup". I've tried to run this test with the flag set to False and it makes no difference

Reply:
It's redundant now. Previously this variable was checked inside of `init_wallet()` which was called by `import_deterministic_coinbase_privkeys()` so we needed to set it to `True` to allow our call of that to work. But since checking `requires_wallet` is moved outside of that, it is no longer needed.

Removed and dropped the commit.